src/openrct2/world/sprite.c

```diff
@@ -767,9 +767,9 @@ void sprite_position_tween_all(float nudge)
             rct_xyz16 posB = _spritelocations2[i];
 
             sprite_set_coordinates(
-                posB.x + (sint16)((posA.x - posB.x) * nudge),
-                posB.y + (sint16)((posA.y - posB.y) * nudge),
-                posB.z + (sint16)((posA.z - posB.z) * nudge),
+                posB.x * alpha + posA.x * (1.0f - alpha),
+                posB.y * alpha + posA.y * (1.0f - alpha),
+                posB.z * alpha + posA.z * (1.0f - alpha),
```

Comment:
Why have you changed the calculations here? Does this now mean you are doing 9 flops per sprite instead of 12 flops per sprite instead of 6?

Also `(1.0f - alpha)` can definitely be calculate outside the loop and re-used.